Feature: add application scopes framework and imp health scope (#367)
oam-dev/rudr: healthscope/README.md

# Health Scope Controller

> Health Scope Controller used for periodically check health scope crd and check the health of all the components related.

Reviewer: ```suggestion
HealthScope controller is used to periodically check the health of all the components and update health scope custom resources.
```

## What will health scope controller do?

> 1. periodically check all component health and update the CR status.

Reviewer: ```suggestion
1. periodically check health status of components and update the HealthScope resource status.
```

2. serve as a http server, to output aggregated health information.

## How to install?

> You could use helm to install it:

Reviewer: ```suggestion
Use helm to install:
```

> ```shell script

Reviewer: ```suggestion
```shell
```

Author: my IDE auto complete `shell` to `shell script `

helm install healthscope ./charts/healthscope
```

## How to use?

> After healthscope was installed by helm charts, you could setup an endpoint to it.

Reviewer: ```suggestion
After healthscope controller is installed by helm charts, setup an endpoint to it.
```

> By default the charts will use ClusterPort, you could use port mapper to make health scope accessible.

Reviewer: ```suggestion
By default the charts will use ClusterPort, we can use port-forward to access healthscope controller.
```

```shell script
export POD_NAME=$(kubectl get pods -l "app.kubernetes.io/name=healthscope,app.kubernetes.io/instance=health" -o jsonpath="{.items[0].metadata.name}")
kubectl port-forward $POD_NAME 8080:80
```

> Then you'll be able to visit http://127.0.0.1:8080 to get health scope status.

Reviewer: ```suggestion
Then we'll be able to visit http://127.0.0.1:8080 to query status of healthscope resource.
```
